packages/abis/DistributionPool.json

```diff
@@ -0,0 +1 @@
ew","type":"function"},{"inputs":[],"name":"claimAllocation","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[],"name":"didCreatorLockTokens","outputs":[{"internalType":"bool","name":"","type":"bool"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"_investor","type":"address"},{"internalType":"uint256","name":"_milestoneId","type":"uint256"}],"name":"getAllocatedAmount","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"_investor","type":"address"}],"name":"getAllocatedTokens","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"_investor","type":"address"}],"name":"getAllocationData","outputs":[{"internalType":"uint256","name":"alreadyAllocated","type":"uint256"},{"internalType":"uint256","name":"allocationFlowRate","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"_investor","type":"address"}],"name":"getClaimedTokens","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"getInvestmentPool","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"getLockedTokens","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"address","name":"_investor","type":"address"}],"name":"getMilestonesWithAllocation","outputs":[{"internalType":"uint256[]","name":"","type":"uint256[]"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"getPercentageDivider","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"pure","type":"function"},{"inputs":[],"name":"getToken","outputs":[{"internalType":"address","name":"","type":"address"}],"stateMutability":"view","type":"function"},{"inputs":[],"name":"getTotalAllocatedTokens","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"contract IInvestmentPool","name":"_investmentPool","type":"address"},{"internalType":"contract IERC20","name":"_projectToken","type":"address"},{"internalType":"uint256","name":"_amountToLock","type":"uint256"}],"name":"initialize","outputs":[],"stateMutability":"payable","type":"function"},{"inputs":[],"name":"lockTokens","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint256","name":"_milestoneId","type":"uint256"},{"internalType":"address","name":"_investor","type":"address"}],"name":"removeTokensAllocation","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[],"name":"withdrawTokens","outputs":[],"stateMutability":"nonpayable","type":"function"}]
```

Comment:
I have a question about ABIs in general - do we keep them in repo or not?

On one hand - ABIs are auto generated, and entirely produced from the repository code. (Artefacts)
Meaning that they must be updated each time contracts change (before each commit effectively).
If it's not done - the repo will have wrong ABIs that don't match contracts.
By that logic  - they don't really belong here.

On the other hand - it's quite convenient to just have them available at all times.

I'd want to hear your input on this.

Reply:
Currently, the subgraph doesn't store all values from the smart contract that the frontend team will use. We will add more and more entities and values to the subgraph over time. The current smart contract that is used by the fronted is a little bit outdated as it uses old function names (they were renamed for better understanding) and return types (after optimization some values now return number and not BigNumber type in typescript). Now I have added only the milestones entity information, which frontend will be able to use (after the branch is merged). To make sure that old values, which are not received from the subgraph, work, we use the old ABI. After all entities and values are stored by the subgraph and the frontend side uses the subgraph for all GET values (not SET), we will need to deploy a new contract (latest version). Then in the subgraph, I will go through all the mappings and change the names or other parts that need to be updated. And because the frontend will already be receiving all information from the subgraph, they won't need to change anything in the frontend part.

That's why I think at this stage storing the ABI is a good idea as it ensures everyone is using the same version of it.